src/cryptography/hazmat/backends/openssl/dh.py

```diff
@@ -149,8 +139,11 @@ def exchange(self, peer_public_key):
         )
 
         if res == -1:
-            errors = self._backend._consume_errors()
-            return _handle_dh_compute_key_error(errors, self._backend)
+            errors_with_text = self._backend._consume_errors_with_text()
```

Comment:
We only know of an invalid public key here, but we want to value error no matter what and this attaches errors_with_text to the ValueError to provide more context.